lib/krane/cluster_resource_discovery.rb

```diff
@@ -17,13 +17,38 @@ def crds
     end
 
     def global_resource_kinds
-      @globals ||= fetch_globals.map { |g| g["kind"] }
+      @globals ||= fetch_resources(only_globals: true).map { |g| g["kind"] }
+    end
+
+    def prunable_resources
+      api_versions = fetch_api_versions
+      fetch_resources.map do |resource|
+        next unless resource['verbs'].include?("delete")
+        version = api_versions.fetch(resource['apigroup'], ['v1']).last
+        [resource['apigroup'], version, resource['kind']].compact.join("/")
+      end.compact
     end
 
     private
 
-    def fetch_globals
-      raw, _, st = kubectl.run("api-resources", "--namespaced=false", output: "wide", attempts: 5,
+    def fetch_api_versions
+      raw, _, st = kubectl.run("api-versions", attempts: 5, use_namespace: false)
+      if st.success?
+        rows = raw.split("\n")
+        rows.each_with_object({}) do |group_version, hash|
+          group, version = group_version.split("/")
```

Comment:
When I run this locally, I see core/v1 is listed as `v1`, which will give us a `v1 => []` entry in this array--it's actually the opposite, since v1 is of course a version not a group. Actually, if we handle that here, I think we can remove the special case above, since there'd be an entry in the hash for the blank api group?